relay-server/src/actors/processor.rs

```diff
@@ -2756,6 +2769,7 @@ impl EnvelopeProcessorService {
             });
 
             self.extract_event(state)?;
+            self.transfer_profile_id(state);
```

Comment:
This is the main change of this PR: The transfer of the profile ID happens _after_ event extraction. The rest of the PR is largely identical with https://github.com/getsentry/relay/pull/2715.